scss/_button-group.scss

```diff
@@ -125,7 +125,7 @@
 
   > .btn:not(:first-child),
   > .btn-group:not(:first-child) {
-    margin-top: -$btn-border-width;
+    margin-top: $btn-group-vertical-not-first-child-margin-top;
```

Comment:
For this case need to consider what to do, because this is more technical solution to keep margins with border and align everything, so probably it will be always `-$btn-border-width`, so solution here can be

`margin-top: calc($btn-border-width * -1)` or `margin-top: $btn-margin-top-align-by-border-width`